### PR TITLE
Ticket/119 implement new service code

### LINKED ISSUE #119: Make API Code Generic and Fix Bad Hook
### Issue description
This ticket is a two parter.
1. `src/services/api/actions/getSearchResults.js` is an action that calls a React hook, which is a no-no. 
2. We need to update the services to mirror https://github.com/NCIOCPL/cgov-react-app-playground/issues/27, and SWS uses 3 APIs and is a good test case. The code changes and approach for this ticket will fix item 1 above, and it should make it so this kind of stuff does not happen again.

> **ESTIMATE** TBD

Nothing should break after this change.

### PR DESCRIPTION
**Pull request by bryanpizzillo**
See https://github.com/NCIOCPL/cgov-react-app-playground/issues/27 for more context.

Closes #119

### CODE REVIEW

File: src/hooks/__tests__/customFetch.test.js [hunk 1]
```diff
@@ -6,11 +6,15 @@ import UseCustomQuerySample from '../samples/UseCustomQuery';
 import { useStateValue } from '../../store/store';
 import MockAnalyticsProvider from '../../tracking/mock-analytics-provider';
 import ErrorBoundary from '../../views/ErrorBoundary';
-import { setDictionaryEndpoint, setLanguage, setSearchEndpoint } from '../../services/api/endpoints';
 import { i18n } from '../../utils';
 
 jest.mock('../../store/store');
 
+// const apiBaseEndpoint = 'http://localhost:3000/api';
```
Comment (seyilonge-nci) [inline src/hooks/__tests__/customFetch.test.js:13]: Any reason you're keeping this block of commented code around, or was that there before and it's the linter making this show up as new code?

File: src/index.js [hunk 2]
```diff
@@ -96,11 +111,36 @@ const initialize = ({
...
+			: null,
+		glossaryEndpoint !== null
+			? replacingRequestInterceptor('glossary-api', {
+					API_HOST: cleanURI(glossaryEndpoint),
+					AUDIENCE: rest.dictionaryAudience,
+					DICTIONARY_NAME: rest.dictionaryName,
+					LANGUAGE: language,
+			  })
+			: null,
+	].filter((item) => item !== null);
```
Comment (seyilonge-nci) [inline src/index.js:137]: 👍

**Review by seyilonge-nci — APPROVED** [review]

File: src/hooks/__tests__/customFetch.test.js (hunk 1 above)
Comment (bryanpizzillo) [reply src/hooks/__tests__/customFetch.test.js]: I must have been drunk. This has been removed.

**Comment by bryanpizzillo** [comment]
Note, for anyone looking, there was an issue in the release mergeback and this commit was changed to 688d0673bdfb8fce0da141b872726003deecb3d1.

**Pull request merged**